crates/chain/src/tx_graph.rs

```diff
@@ -794,7 +794,15 @@ impl<A: Anchor> TxGraph<A> {
         txid: Txid,
     ) -> Result<Option<ChainPosition<&A>>, C::Error> {
         let (tx_node, anchors, last_seen) = match self.txs.get(&txid) {
-            Some(v) => v,
+            Some(val) => {
+                let (_, anchors, last_seen) = val;
+                if anchors.is_empty() && last_seen.is_none() {
+                    // We don't consider unanchored, unseen transactions
+                    // to have a chain position.
+                    return Ok(None);
+                }
+                val
+            }
```

Comment:
I like the idea of moving this logic into `try_get_chain_position`. However, for transactions with `last_seen` as `None`, but with an anchor that anchors the transaction in the best chain, wouldn't the transaction be in the best chain?

My understanding of `last_seen` is that a value of `None` signals that it has not been seen in the mempool. However, this fact shouldn't invalidate the fact that the transaction can be confirmed in the best chain?

Reply:
Should we rename this method to `try_get_canonical_position`?

Reply:
I think having a chain position should imply the tx is canonical, so I think these are synonymous.